pipelines/tools-staging-prod-infra.yaml

```diff
@@ -0,0 +1,422 @@
+
+groups:
+- name: provision
+  jobs:
+  - terraform
+  - bootstrap-tools-cluster
+  - apply-addons-tools-cluster
+  - bootstrap-staging-cluster
+  - apply-addons-staging-cluster
+  - bootstrap-prod-cluster
+  - apply-addons-prod-cluster
+- name: destroy
+  jobs:
+  - destroy-tools-cluster
+  - destroy-staging-cluster
+  - destroy-prod-cluster
+
+
+
+
+
+task_image_resource: &task_image_resource
+  type: docker-image
+  source: {repository: "govsvc/task-toolbox", tag: "1.0.0"}
+
+
+apply_addons_task: &apply_addons_task
+  platform: linux
+  image_resource: *task_image_resource
+  params: &apply_addons_task_params
+    ACCOUNT_ROLE_ARN: ((account-role-arn))
+    ACCOUNT_NAME: ((account-name))
+    CLUSTER_NAME:
+    TERRAFORM_BUCKET: cd-gsp-private-qndvvc
+    TERRAFORM_REGION: eu-west-2
+    TF_VAR_aws_account_role_arn: ((account-role-arn))
+  run:
+    path: /bin/bash
+    args:
+    - -eu
+    - -c
+    - |
+      echo "fetching kubeconfig from terraform state..."
+      jq -r '."admin-kubeconfig"' cluster-outputs/metadata > kubeconfig
+      export KUBECONFIG=$(pwd)/kubeconfig
+      echo "generating kubeyaml addons with terraform..."
+      cd "gsp-teams/terraform/accounts/${ACCOUNT_NAME}/clusters/${CLUSTER_NAME}"
+      terraform init \
+        -backend-config="bucket=${TERRAFORM_BUCKET}" \
+        -backend-config="key=${ACCOUNT_NAME}/clusters/${CLUSTER_NAME}.tfstate" \
+        -backend-config="region=${TERRAFORM_REGION}" \
+        -backend=true \
+        -force-copy \
+        -get=true \
+        -input=false
+      terraform workspace select "${ACCOUNT_NAME}"
+      terraform apply -auto-approve
+      echo "applying kubeyaml..."
+      eval $(aws-assume-role $ACCOUNT_ROLE_ARN)
+      until kubectl apply -R -f addons/; do
+        echo "apply failed retrying..."
```

Comment:
We should add something to this `echo` that it might loop forever in case someone looking in doesn't know context?

Reply:
there is a 30min timeout on the concourse task to fail it ... but yeah being more informative sounds like a good plan